test/upgrade_test.go

```diff
@@ -0,0 +1,180 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/test/parse"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+// TestUpgradeAgainstPreviousVersion is a light-weight upgrade test for examining
+// the existing functionalities against preivous server version to avoid backward
+// incompatibility. It creates a pipelineRun with a basic TaskRun and a basic
+// Pipeline and examines the events and runs created are successful.
```

Comment:
The docstring here is a good description of what this PR is doing. However, it's not really a description of what this test is doing. This test runs against whatever server version is installed on the cluster, and it's the logic in e2e-tests-upgrade.sh that makes it an "upgrade" test. Could you update the docstring and test name to reflect this?

Reply:
Thanks for updating the docstring; don't forget to update the test name!

Reply:
Done! Thanks for the reminder